pkg/autohealing/controller/controller.go

```diff
@@ -371,6 +379,10 @@ func (c *Controller) startWorkerMonitor(wg *sync.WaitGroup) {
 	log.V(3).Info("Finished checking worker nodes.")
 }
 
+func (c *Controller) updateClusterHealthStatus() {
+	c.provider.UpdateHealthStatus(totalUnhealthyNodes)
```

Comment:
We don't need an extra function here.

Why do we need to call UpdateHealthStatus function three times (master+worker+total)?

Reply:
Yep, we should be able to drop the `updateClusterHealthStatus()` for Controller and then call the provider's `UpdateClusterHealthStatus()` directly.

The `UpdateHealthStatus()` could be dropped from the Repair()  function and only keep the one in Controller. I will test it and fix it in the next commit.